src/go/rpk/pkg/publicapi/dataplane.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package publicapi
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"connectrpc.com/connect"
+)
+
+// DataPlaneClientSet holds the respective service clients to interact with
+// the data plane endpoints of the Public API.
+type DataPlaneClientSet struct {
+	Transform transformServiceClient
+}
+
+// NewDataPlaneClientSet creates a Public API client set with the service
+// clients of each resource available to interact with this package.
+func NewDataPlaneClientSet(host, authToken string, opts ...connect.ClientOption) (*DataPlaneClientSet, error) {
+	if host == "" {
+		return nil, fmt.Errorf("dataplane host is empty")
+	}
+	opts = append([]connect.ClientOption{
+		connect.WithInterceptors(
+			newAuthInterceptor(authToken), // Add the Bearer token.
+			newLoggerInterceptor(),        // Add logs to every request.
+		),
+	}, opts...)
+
+	httpCl := &http.Client{Timeout: 15 * time.Second}
```

Comment:
Do we even need a timeout, or can we rely on the input context? I don't think we have a relevant global override to use yet (just checked), 15s ideally is fine...

Reply:
We can rely on the input context (i.e, CTRL+C if it's hanging). This is only used for the `deploy` command which can be slow if you are uploading a heavy transform and not having a way to override the timeout is bad.

I'm removing it.